cardboardlint/common.py

```diff
@@ -0,0 +1,188 @@
+"""Collection of classes and methods shared between different linters."""
+import os
+from fnmatch import fnmatch
+import subprocess
+
+
+class Message(object):
+    """Error message and meta information.
+
+    This class contains all the machinery that the trapdoor driver uses to check for new messages.
+    When a context is available, it is used to compare two messages instead of line numbers. If not,
+    the line numbers are used. Line numbers are a relatively poor descriptor for assessing if a
+    message is new. For example, lines may have been inserted or removed, which changes the line
+    numbers without actually changing any of the code.
+    """
+
+    def __init__(self, filename, lineno, charno, text, context=None):
+        """Initialize a message.
+
+        Parameters
+        ----------
+        filename : str
+            The filename from which the message is reported.
+        lineno : int (or None)
+            The line number at which the error/problem is reported.
+            None if no error/problem is reported.
+        charno : int (or None)
+            The character position at which the error/problem is reported.
+            None if no error/problem is reported.
+        text : str
+            A description of the error/problem.
+        context : str
+            A string that (almost) uniquely identifies the location of the error without using line
+            numbers.
+        """
+        if lineno is not None and not isinstance(lineno, int):
+            raise TypeError('`lineno` must be an integer or None')
+        if charno is not None and not isinstance(charno, int):
+            raise TypeError('`charno` must be an integer or None')
+        self.filename = filename
+        self.lineno = lineno
+        self.charno = charno
+        self.text = text
+        self.context = context
+
+    # FIXME: is this needed? __eq__ and __has__ were removed (not sure )
```

Comment:
This is still going to be useful to sort messages before printing. See other comment below.